lib/src/git.rs

```diff
@@ -268,6 +268,98 @@ impl CreateWorktreeError {
+    // OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+    // ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+    // OTHER DEALINGS IN THE SOFTWARE.
+    //
+
+    #[cfg(target_os = "windows")]
+    mod windows {
+        use std::os::windows::ffi::OsStrExt as _;
+        use std::path::MAIN_SEPARATOR;
+
+        use super::*;
+
+        // Workaround for Windows. There is no way to extract raw byte sequence from
+        // `OsStr` (in `Path`). And `OsStr::to_string_lossy` may cause extra
+        // heap allocation.
+        pub(crate) fn ends_with_main_sep(p: &Path) -> bool {
+            p.as_os_str().encode_wide().last() == Some(MAIN_SEPARATOR as u16)
+        }
+    }
+    #[cfg(not(target_os = "windows"))]
+    fn to_slash_lossy(path: &Path) -> Cow<'_, str> {
+        path.to_string_lossy()
+    }
+    #[cfg(target_os = "windows")]
+    fn to_slash_lossy(path: &Path) -> Cow<'_, str> {
+        use std::path::Component;
+        use std::path::Prefix;
+
+        let mut buf = String::new();
+        for c in path.components() {
+            match c {
+                Component::RootDir => { /* empty */ }
+                Component::CurDir => buf.push('.'),
+                Component::ParentDir => buf.push_str(".."),
+                Component::Prefix(prefix_component) => {
+                    match prefix_component.kind() {
+                        Prefix::Disk(disk) | Prefix::VerbatimDisk(disk) => {
+                            if let Some(c) = char::from_u32(disk as u32) {
+                                buf.push(c);
+                                buf.push(':');
+                            }
+                        }
+                        Prefix::UNC(host, share) | Prefix::VerbatimUNC(host, share) => {
+                            // Write it as non-verbatim but with two forward slashes // instead of
+                            // \\. I think this sounds right? https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/unc-path-recognition-unix
+                            buf.push_str("//");
+                            buf.push_str(&host.to_string_lossy());
+                            buf.push_str("/");
+                            buf.push_str(&share.to_string_lossy());
+                        }
+                        // Just ignore it and hope for the best?
+                        Prefix::Verbatim(_) => {}
+                        Prefix::DeviceNS(_) => {}
+                    }
+                    // C:\foo is [Prefix, RootDir, Normal]. Avoid C://
+                    continue;
+                }
+                Component::Normal(s) => buf.push_str(&s.to_string_lossy()),
+            }
+            buf.push('/');
+        }
+
+        if !windows::ends_with_main_sep(path) && buf != "/" && buf.ends_with('/') {
+            buf.pop(); // Pop last '/'
+        }
+
+        Cow::Owned(buf)
```

Comment:
@martinvonz Does this need to be a dep for Google, since its under a separate license?

Reply:
Thanks for asking. No, but `lib/LICENSE` should be updated to say that the MIT license applies to this file. See https://github.com/libgit2/libgit2/pull/6413 and https://github.com/mitsuhiko/sha1-smol/pull/47 for reference.

Reply:
cc @ilyagr ref https://github.com/martinvonz/jj/pull/4188 re path-slash

Reply:
Moved to lib/LICENSE, but this will be in a followup PR.